File: SORTING/Intersection of two sorted arrays.py

class Solution:
#Function to return a list containing the intersection of two arrays.

    def printIntersection(self,a, b, n, m):
     
        # code here
        ans = list()
        aa= set(a)
        bb = set(b)
        res = aa.intersection(bb)
        if len(res) ==0:
            return -1
        for i in res:
            ans.append(i)
        ans.sort()
        return ans
            
            

#{ 
#  Driver Code Starts
#Initial Template for Python 3
import io
import sys

#Contributed by : Nagendra Jha

if __name__ == '__main__':
    test_cases = int(input())
    for cases in range(test_cases) :
        n,m = map(int,input().strip().split())
        a = list(map(int,input().strip().split()))
        b = list(map(int,input().strip().split()))
        ob=Solution()
        l = ob.printIntersection(a,b,n,m)
        print(*l)
# } Driver Code Ends

File: SLIDING WINDOW/IntersectionOfTwoSortedArrays.cpp
vector<int> Solution::intersect(const vector<int> &A, const vector<int> &B) {
    vector<int> res;
    int i=0;
    int j=0;
    while(i<A.size() && j<B.size())
    {
        if(A[i]<B[j])
        i++;
        if(A[i]>B[j])
        j++;
        if(A[i]==B[j])
        {
            res.push_back(A[i]);
            i++;
            j++;
        }
    }
    return res;
}
